server/src/test-fast/java/com/thoughtworks/go/server/web/DefaultHeadersFilterTest.java

```diff
@@ -16,44 +16,73 @@
 
 package com.thoughtworks.go.server.web;
 
+import com.thoughtworks.go.server.service.support.toggle.FeatureToggleService;
+import com.thoughtworks.go.util.SystemEnvironment;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 import org.mockito.Mock;
 
 import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
 import javax.servlet.ServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.verifyNoMoreInteractions;
-import static org.mockito.Mockito.when;
+import java.io.IOException;
+
+import static org.mockito.Mockito.*;
 import static org.mockito.MockitoAnnotations.initMocks;
 
 public class DefaultHeadersFilterTest {
 
+    private static final String ENABLE_HSTS_HEADER = "gocd.enable.hsts.header";
     @Mock
     private HttpServletResponse response;
     @Mock
     private FilterChain chain;
     @Mock
     private ServletRequest request;
+    @Mock
+    private FeatureToggleService featureToggleService;
     private DefaultHeadersFilter filter;
+    private SystemEnvironment systemEnvironment;
 
     @Before
     public void setUp() throws Exception {
         initMocks(this);
         filter = new DefaultHeadersFilter();
+        systemEnvironment = new SystemEnvironment();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        systemEnvironment.clearProperty(ENABLE_HSTS_HEADER);
     }
 
     @Test
     public void shouldAddDefaultHeaders() throws Exception {
+        systemEnvironment.setProperty("gocd.enable.hsts.header", "true");
+        filter.doFilter(request, response, chain);
+
+        verify(response).isCommitted();
+        verify(response).setHeader("X-XSS-Protection", "1; mode=block");
+        verify(response).setHeader("X-Content-Type-Options", "nosniff");
+        verify(response).setHeader("X-Frame-Options", "SAMEORIGIN");
+        verify(response).setHeader("X-UA-Compatible", "chrome=1");
+        verify(response).setHeader("Strict-Transport-Security", "31536000");
```

Comment:
Is it `Strict-Transport-Security: 31536000` or `Strict-Transport-Security: max-age=31536000 `?

Reply:
From https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security#Syntax

> Syntax:
> 
> ```
> Strict-Transport-Security: max-age=<expire-time>
> Strict-Transport-Security: max-age=<expire-time>; includeSubDomains
> Strict-Transport-Security: max-age=<expire-time>; preload
> ```

Reply:
Also — should we allow configuring the params?

Reply:
We're going with four options: `-Dgocd.enable.hsts.header=true`, `-Dgocd.hsts.header.maxage=[value]` (defaults to 1 year), `-Dgocd.hsts.header.includesubdomains=true` and `-Dgocd.hsts.header.preload=true`. The user needs to use just the first one to enable the header, and the rest can be used for further customization